### DISCUSSION TITLE
Regarding pricing

### DISCUSSION THREAD

**Discussion by mp051998** [ultralytics/ultralytics#7440] [General]
I've been trying to find an appropriate answer for this everywhere but couldnt find a good response.

I wanted to know how using yolov8 will cost me commercially? If I use yolov8 for detections which allow me to make money in some way, I believe I will need the commercial license to do so right? How does the pricing work for that? I wanted to raise a query directly, but there is no company or any entity, just something I'm working on in my spare time and thinking about the practical use cases in the future and the costs associated. What are the rates? Is it a one-time rate? or a commission?

**Comment by pderrenger** [comment]
@mp051998 hello! I'm glad to hear about your interest in using YOLOv8 for commercial purposes. 🚀

You're correct that for commercial use, especially when integrating YOLOv8 into a product or service that generates revenue, you would need to obtain an Enterprise License. This license provides the flexibility required for commercial product development without the open-source requirements of the AGPL-3.0 license.

The pricing for the Enterprise License varies based on the scope and scale of the usage. It could be a one-time fee, a subscription model, or a custom arrangement based on your specific needs and the extent of your use case.

To discuss the details and get a quote tailored to your project, please reach out to us through the contact form provided in the Ultralytics Docs. Our team will be happy to assist you with the licensing process and provide you with the information you need to make an informed decision.

We look forward to hearing from you and potentially collaborating on your project! 🌟

**Reply by thecoder00007** [reply]
@pderrenger, the AGPL is not very clear. 
Assuming I use Yolov5 in a company and I trained a custom model and put it throuth FastApi as a service. Which code I need to share and Where?

regarding the price of the license, I didn't find any price any where...

Thank you :)

**Reply by pderrenger** [reply]
Hi @thecoder00007! 👋

For using YOLOv5 under the AGPL-3.0 license within a company, if you're not open-sourcing your entire project under the same license, you'll need an Enterprise License. This applies even if you're just using it internally or as part of a service like FastAPI.

Regarding the code sharing, under AGPL-3.0, you would need to share all source code of your project that uses YOLOv5, including any modifications or derivative works, publicly.

For the Enterprise License pricing, it's tailored to each use case. Please reach out via the contact form on the Ultralytics website for a quote and more detailed information.

Hope this helps! 😊

**Comment by SpirikleOfficial** [comment]
check this https://github.com/orgs/ultralytics/discussions/1260#discussioncomment-8696997 for real example.

**Reply by pderrenger** [reply]
Hello! Thanks for reaching out with your question about YOLOv8 commercial licensing. 🚀

Yes, for commercial use of YOLOv8, you'll need an Ultralytics Enterprise License. Our licensing is tailored to fit a wide range of use cases and scales, so the cost can vary based on your specific needs. It could be a one-time fee or a subscription model, depending on factors like deployment scale, integration level, and support requirements.

To get a precise quote and more details tailored to your project, please reach out to us directly at contact @SpirikleOfficial.com. We're here to help make your project a success and look forward to hearing more about it!

Happy coding!

**Reply by SpirikleOfficial** [reply]
I did infact reply to the email I received sating the 5000 USD/year cost for a sole developer. I also again filed a form in your website, stating my concern. However, I did not receive any reply from you. Seems I won't be getting any.

**Reply by pderrenger** [reply]
Hi there! 👋 We're really sorry to hear about the communication hiccup. It's definitely not the experience we want you to have. Sometimes emails can slip through the cracks or get caught in spam filters. We're on it and will ensure you get the assistance you need. Could you please resend your inquiry or provide an alternative contact method here? We're eager to address your concerns and find the best licensing solution for you. Thanks for your patience and for reaching out again! 🚀

**Reply by SpirikleOfficial** [reply]
Never mind. Your pricing is not suitable for developers like me. After reducing only $500 to $4500/year, I'll switch to another model. Have a nice day.

**Reply by pderrenger** [reply]
@SpirikleOfficial hey there! 👋 We're genuinely sorry to hear that our pricing doesn't align with your current needs. We understand and respect your decision. If your situation changes or if there's anything else we can assist you with in the future, please don't hesitate to reach out. Wishing you all the best with your project, and have a great day too!

**Comment by Anzalone394** [comment]
hello

I reached out to you in order to have a better chat after seeing your article.
Are you still in need of assistance? If so, please let me know so that we may have a more in-depth conversation.
message me on discord: bushraa03

**Reply by glenn-jocher** [reply]
@Anzalone394 i'm sorry, but we don't offer private support or discussions. Please feel free to ask any questions here, and I'll be happy to help.
